feat(cloudflare): add support for cloudflare's image cdn

nicksrandall opened the pull request:
Fixes: #1

nicksrandall commented:
I'm excited to delete the custom, 100+ line image component in my app and replace it with this -- I just need it to support Cloudflare :)

If it's not too much trouble, could you please push a new version to NPM of this and `unpic-img` when (if?) this PR is merged?

ascorbic commented on src/transformers/cloudflare.ts:
```diff
@@ -0,0 +1,111 @@
+  UrlTransformer,
+} from "../types.ts";
+
+// Thanks Colby!
```
Probably don't need to thank Colby for this one!

ascorbic commented on src/transformers/cloudflare.ts:
```diff
@@ -0,0 +1,111 @@
+    path
+  }: CloudflareParams,
+): string => {
+  console.log('trans', transformations);
```
Left this in

ascorbic reviewed (changes requested):
Thanks for this! Are there any default domains that it can auto-detect, or is it always the user's domain? Could you share a real example image to test it, preferably one from the Cloudflare docs.

Could you also add it to the README, but with a note saying that it's not auto-detected and needs to be manually specified. Thanks again!

nicksrandall commented:
I'll get your feedback implemented. 

Regarding auto detection by domain, that is not possible with cloudflare because it always uses the users domain. That said, it would be possible to detect by a pathname prefix (`/cdn-cgi/image`). Do you think its worth adding that logic (similar to how you currently do subdomains)?

nicksrandall commented:
@ascorbic I addressed your feedback. I wasn't able to find an image in the docs that works but I did change the URL to match what is in the docs. 

Also, I was able to add Cloudflare to the detection function by looking for a specific path prefix. Let me know if you think that is acceptable.

ascorbic commented on src/transformers/cloudflare.test.ts:
```diff
@@ -0,0 +1,39 @@
+import { CloudflareParams, parse, transform } from "./cloudflare.ts";
+
+const img =
+  "https://assets.example.comeight=64,f=auto,background=red/uploads/avatar1.jpg"
```
I think something's wriong with this URL!

nicksrandall replied:
Fixed

ascorbic commented on src/transformers/cloudflare.test.ts:
```diff
@@ -0,0 +1,39 @@
+  assertEquals(parsed, expected);
+});
+
+Deno.test("cloudfalre transformer", async (t) => {
```
```suggestion
Deno.test("cloudflare transformer", async (t) => {
```

ascorbic reviewed (commented):
Good idea with the path matching. I think there are some others that we could use that approach for. Just a couple of typos left. I would really like if we can find at least one valid URL that can be used to demo this, so it can be added to the playground in examples.json. Otherwise I've no way of being sure it actually works. Comparing it to the docs only goes so far.

ascorbic commented:
The tests are failing because of that broken url

nicksrandall commented:
Fixed the test with a generic image from my app. Hopefully that is good enough?

ascorbic reviewed (approved):
Perfect! Thanks for your help
